web/firefoxcom.js

```diff
@@ -308,19 +308,53 @@ class FirefoxScripting {
 }
 
 class MLManager {
-  #enabled = new Map();
+  #enabled = null;
 
-  constructor({ enableAltText }) {
-    this.#enabled.set("altText", enableAltText);
+  eventBus = null;
+
+  constructor(options) {
+    this.enable({ ...options, listenToProgress: false });
   }
 
-  isEnabledFor(name) {
-    return this.#enabled.get(name);
+  async isEnabledFor(name) {
+    return !!(await this.#enabled.get(name));
   }
 
   guess(data) {
     return FirefoxCom.requestAsync("mlGuess", data);
   }
+
+  enable({ enableAltText, listenToProgress }) {
+    if (enableAltText) {
+      this.#loadAltTextEngine(listenToProgress);
+    }
+  }
+
+  async #loadAltTextEngine(listenToProgress) {
+    if (this.#enabled.has("altText")) {
```

Comment:
Again, won't this throw?
```suggestion
    if (this.#enabled?.has("altText")) {
```